Import plone.restapi docs

stevepiercy opened the pull request:
Depends on merge of https://github.com/plone/plone.restapi/pull/1337. See also https://github.com/plone/plone.restapi/issues/1326

stevepiercy committed:
Pull in plone.restapi docs

stevepiercy committed:
Use correct path to plone.restapi docs source

stevepiercy committed:
Add requirements from plone.restapi

stevepiercy committed:
Generalize how to work with external packages documentation

stevepiercy committed:
Update submodules/plone.restapi tip

stevepiercy committed:
git commit -m "Update submodules/plone.restapi tip"

mister-roboto commented:
@stevepiercy thanks for creating this Pull Request and helping to improve Plone!

TL;DR: Finish pushing changes, pass all other checks, then paste a comment:
```
@jenkins-plone-org please run jobs
```

To ensure that these changes do not break other parts of Plone, the Plone test suite matrix needs to pass, but it takes 30-60 min.  Other CI checks are usually much faster and the Plone Jenkins resources are limited, so when done pushing changes and all other checks pass either [start all Jenkins PR jobs yourself](https://jenkinsploneorg.readthedocs.io/en/latest/run-pull-request-jobs.html#run-a-pull-request-job), or simply add the comment above in this PR to start all the jobs automatically.

Happy hacking!

stevepiercy committed:
Include plone.restapi docs into main Plone docs. Look, Ma! No conversion to MyST needed!

stevepiercy committed:
Correct help for deps target

stevepiercy committed:
Add sphinx extensions used by plone.restapi to Sphinx configuration

stevepiercy committed:
Exclude CONTRIBUTORS.rst and plone.restapi ideas docs from the build
Use new symlink to pull in http request and response examples
Update the path plone.restapi docs to use new symlink

stevepiercy committed:
Tweak syntax for symlink to restapi

stevepiercy committed:
Update the symlinks for git to ignore

stevepiercy committed:
Update submodules/plone.restapi tip

stevepiercy committed:
Exclude paths in plone.restapi from being scanned by Sphinx for source docs files

stevepiercy committed:
Update symlink to plone.restapi package instead of just its docs.
This will allow the http-request tests to build for both the project's and main documentation.

stevepiercy committed:
Add html_meta values

stevepiercy committed:
Update submodule to supported and secure protocol

stevepiercy committed:
Tell Netlify to clone plone.restapi

stevepiercy committed:
Update submodules/volto and submodules/plone.restapi tip

stevepiercy committed:
Remove obsolete TODO, add link to exlude from plone.restapi

stevepiercy committed:
Update submodules/plone.restapi tip

stevepiercy committed:
Merge branch '6-dev' into import-restapi

stevepiercy committed:
Ignore link with anchor

ksuess commented on .gitmodules:
```diff
@@ -2,3 +2,7 @@
 	branch = master
+[submodule "submodules/plone.restapi"]
+	path = submodules/plone.restapi
+	url = https://github.com/plone/plone.restapi.git
```
Check https://github.com/plone/plone.restapi/pull/1337#issuecomment-1074195352. Then this could be changed to  branch master. 
I' ll continue reviewing afterwards, OK?

stevepiercy replied:
I need to keep it as `plone6docs` until it is merged into `master`, else it won't build preview on Netlify. I will change it as soon as it is merged.

ksuess commented on docs/conf.py:
```diff
@@ -80,6 +81,8 @@
     r"https://github.com/plone/plone.volto/blob/6f5382c74f668935527e962490b81cb72bf3bc94/src/kitconcept/volto/upgrades.py#L6-L54",
     r"https://github.com/tc39/proposals/blob/HEAD/finished-proposals.md#finished-proposals",
+    r"https://coveralls.io/repos/github/plone/plone.restapi/badge.svg\?branch=master",  # plone.restapi
+    r"https://github.com/plone/plone.restapi/blob/dde57b88e0f1b5f5e9f04e6a21865bc0dde55b1c/src/plone/restapi/services/content/add.py#L35-L61",  # plone.restapi
```
this is a renaming or move?

stevepiercy replied:
The coveralls link sends an unpredictable response, redirecting the request to an SVG that indicates the percentage of coverage. Since that number can change, I decided to exclude it.

linkcheck 404s on GitHub anchors in links, so I exclude it.

ksuess commented on docs/contributing/index.md:
```diff
@@ -170,16 +170,16 @@ This section describes how to make contributions to files in the `plone/document
+(contributing-editing-external-package-documentation-label)=
 
-### Editing Volto documentation
+### Editing external package documentation
```
Same for other mentionings of external package Volto:
Rplace "Volto" by "external package"

![grafik](https://user-images.githubusercontent.com/1144737/159341487-6a772b11-0482-4978-949a-0627cfe02bec.png)

stevepiercy replied:
Done in 5f93f5533ca5d77925e937bc71863476258c23cc

ksuess reviewed (changes requested):
great job^2!

stevepiercy committed:
Include plone.restapi throughout contributing/index.md
